src/Engines/QueryBuilderEngine.php

```diff
@@ -102,20 +102,33 @@ public function filtering()
     {
         $this->query->where(
             function ($query) {
-                $keyword = $this->setupKeyword($this->request->keyword());
+                $globalKeyword = $this->setupKeyword($this->request->keyword());
+                $queryBuilder = $this->getQueryBuilder($query);
+
                 foreach ($this->request->searchableColumnIndex() as $index) {
                     $columnName = $this->getColumnName($index);
-
+                    // check if custom column filtering is applied
                     if (isset($this->columnDef['filter'][$columnName])) {
-                        $method     = Helper::getOrMethod($this->columnDef['filter'][$columnName]['method']);
-                        $parameters = $this->columnDef['filter'][$columnName]['parameters'];
-                        $this->compileColumnQuery(
-                            $this->getQueryBuilder($query),
-                            $method,
-                            $parameters,
-                            $columnName,
-                            $keyword
-                        );
+                        $columnDef = $this->columnDef['filter'][$columnName];
+                        // check if global search should be applied for the specific column
+                        $applyGlobalSearch = count($columnDef['parameters']) == 0 || end($columnDef['parameters']) !== false;
+                        if (!$applyGlobalSearch) {
+                            continue;
+                        }
+
+                        if ($columnDef['method'] instanceof Closure) {
+                            $whereQuery = $queryBuilder->newQuery();
+                            call_user_func_array($columnDef['method'], [$whereQuery, $this->request->keyword()]);
+                            $queryBuilder->addNestedWhereQuery($whereQuery);
```

Comment:
Indeed. The best thing is we will be able to group multiple where's in the nested where group (and do some more advanced things in where part) easly:
```php
       ->filterColumn('id', function($query, $keyword) {
             $query->where('user_id', $keyword)->where('type', 'customer');
       }, false)  // override a global search (disable) for the custom filter 
```

p.s. just a thought, maybe it would make sense to give a globalKeyword as a third Closure parameter?